src/antelope/stores/utils/text-utils.ts

```diff
@@ -36,12 +36,32 @@ export function truncateAddress(address: string): string {
  */
 export function getShapedNftName(name: string, id: string): string {
     let shapedName = name;
-    if (name.includes(id)) {
-        shapedName = name.replace(id, '');
+    const idAtTheEndRegex = new RegExp(` #?${id}$`);
```

Comment:
this logic was not robust enough before; if there was any match for the ID, it would be replaced. Now the ID is only replaced if it follows the pattern NAME[space][optional #]ID